core/domain/classifier_domain.py

```diff
@@ -387,37 +355,32 @@ def validate(self):
                     'Expected answers to be a list, received %s' %
                     grouped_answers['answers'])
 
-        # Classifier data can be either None (before its stored) or a dict.
-        if not isinstance(self.classifier_data, dict) and self.classifier_data:
-            raise utils.ValidationError(
-                'Expected classifier_data to be a dict|None, received %s' % (
-                    self.classifier_data))
-
-        if not isinstance(self.data_schema_version, int):
+        if not isinstance(self.algorithm_version, int):
             raise utils.ValidationError(
-                'Expected data_schema_version to be an int, received %s' %
-                self.data_schema_version)
+                'Expected algorithm_version to be an int, received %s' %
+                self.algorithm_version)
 
 
 class TrainingJobExplorationMapping(python_utils.OBJECT):
     """Domain object for a job-exploration mapping model.
 
     A job-exploration mapping is a one-to-one relation between the
     attributes in an exploration to the training job model for the classifier it
-    needs to use. The mapping is from <exp_id, exp_version, state_name> to the
-    job_id.
+    needs to use. The mapping is from <exp_id, exp_version, state_name,
+    algorithm_id> to the job_id.
 
     Attributes:
         exp_id: str. ID of the exploration.
         exp_version: int. The exploration version at the time the corresponding
             classifier's training job was created.
         state_name: str. The name of the state to which the classifier
             belongs.
-        job_id. str. The unique ID of the training job in the
-            job-exploration mapping.
+        algorithm_id_to_job_id_map. dict. Mapping of algorithm IDs to
```

Comment:
Explain why multiple algorithms may be needed for a state. (Maybe be clear that these are all meant to be "current" but used for different contexts.)

Reply:
Added more explanation